File: model-client/src/main/java/org/modelix/model/lazy/CLElementRef.java
package org.modelix.model.lazy;




public class CLElementRef {
  private long id;

  public CLElementRef(long id) {
    this.id = id;
  }

  public long getId() {
    return id;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || this.getClass() != o.getClass()) {
      return false;
    }

    CLElementRef that = (CLElementRef) o;
    if (id != that.id) {
      return false;
    }

    return true;
  }

  @Override
  public int hashCode() {
    int result = 0;
    result = 31 * result + (int) (id ^ (id >> 32));
    return result;
  }
}

File: model-client/src/main/kotlin/org/modelix/model/lazy/CLElementRef.kt
package org.modelix.model.lazy

class CLElementRef(val id: Long) {

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o == null || this.javaClass != o.javaClass) {
            return false
        }
        val that = o as CLElementRef
        return if (id != that.id) {
            false
        } else true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (id xor (id shr 32)).toInt()
        return result
    }
}
